### PR TITLE
Extract archive on macos

### PR DESCRIPTION
**Pull request by bensejas**
Adding support to extract an archive file on macOS agents. `bsdtar` is the default `tar` on macOS and its command options differs from `gnu-tar` which is the default on ubuntu.

### CODE REVIEW

**Review by arromer — APPROVED** [review]
LGTM 👌

File: hooks/checkout
Comment (RayBrokeSomething) [inline hooks/checkout:165]:
```suggestion
    aws s3 cp "${s3_url}/${checkout}" "${PWD}/${checkout}" --silent
```

Comment (RayBrokeSomething) [inline hooks/checkout:165]:
```suggestion
    aws s3 cp "${s3_url}/${checkout}" "${PWD}/${checkout}" --silent
```

File: hooks/checkout
Comment (RayBrokeSomething) [inline hooks/checkout:165]:
```suggestion
    aws s3 cp "${s3_url}/${checkout}" "${PWD}/${checkout}" --quiet
```

oops messed it up

Comment (RayBrokeSomething) [reply hooks/checkout:165]: Actually can fully remove this, it didn't do what I thought it would do... stop logging the download percentage.

**Comment by bensejas** [comment]
@arromer I'm going to need another review. I added 2 changes after the initial approval:
1. Removed `pushd .. > /dev/null` and `popd` to stay within the working directory when extracting the S3 archive.
2. Attempt to change the directory to the first child of the extracted archive when the extracted archive is not a git repository. The next steps expect the current directory to be a git respository.

File: hooks/checkout
Comment (arromer) [inline hooks/checkout:175]: when would this be required?
I don't think this is going to work since the repo would then be located in a different directory to where buildkite will try to use.

**Review by arromer — APPROVED** [review]

**Pull request merged**
